### DISCUSSION TITLE
Next.js routing/apihandling

### DISCUSSION THREAD

**Discussion by Kingjesse89123** [vercel/next.js#44579] [Help]
This is a next.js handler function for dealing with api requests on our server i need to access a variable between request methods any help will be muchly appreciated
export default function handler(req, res) {
    if(req.method ===  'POST') {
        var hi = req.body
    }
    if (req.method === 'GET') {
        console.log(hi)
    }
}

**Comment by icyJoseph** [comment]
Hi,

So if I understand correctly you want to have:

- Request 0, arrives as GET and you do some work
- Request 1, arrives, say, as POST, and based on the work done to Request 0, do some work to Request 1

Is this what you want to achieve?

The problem is that REST is stateless, and each request is a whole-unit transaction, that has no dependency on the previous. Naturally, if you have one server handling all requests, the server can set up "globals" or rather, a common JavaScript context, for all requests, at the risk of leaking data and memory.

What's more interesting is that, in a serverless architecture, such as Next.js deployed on Vercel, the handler runs once, and if its not needed anymore, it will be terminated, so any notion of a common JavaScript context becomes extremely difficult to achieve. There's simply no guarantee that two successive requests will see the same handler. 

What you could do is sign a cookie, or payload as result of Request 0, and include it in the response. When Request 1 arrives, it has to carry that payload/cookie, and you'd need to verify that it is not tampered with, and then proceed to read it. That's a good use case of JWTs. There's a whole area of technology trying to solve this issue in a good and secure way.
